restli-common/src/main/java/com/linkedin/restli/internal/common/URIMaskUtil.java

```diff
@@ -121,161 +121,161 @@ public static MaskTree decodeMaskUriFormat(StringBuilder toparse) throws Illegal
     ParseState state = ParseState.PARSE_FIELDS;
 
     DataMap result = new DataMap();
-    Deque<DataMap> stack = new ArrayDeque<DataMap>();
+    Deque<DataMap> stack = new ArrayDeque<>();
     stack.addLast(result);
+    int position = 0;
 
-    while (toparse.length() > 0)
+    while (position < toparse.length())
```

Comment:
Also the method can take the string as argument to avoid making a copy using StringBuilder